Resources/views/Menu/sonata_menu.html.twig

```diff
@@ -0,0 +1,136 @@
+{% macro attributes(attributes) %}
+{% for name, value in attributes %}
+    {%- if value is not none and value is not sameas(false) -%}
+        {{- ' %s="%s"'|format(name, value is sameas(true) ? name|e : value|e)|raw -}}
+    {%- endif -%}
+{%- endfor -%}
+{% endmacro %}
+
+{% block compressed_root %}
+{% spaceless %}
+{{ block('root') }}
+{% endspaceless %}
+{% endblock %}
+
+{% block root %}
+{% set listAttributes = item.childrenAttributes|merge({'class': 'sidebar-menu'}) %}
+{% set request        = item.getExtra('request') %}
+{{ block('list') -}}
+{% endblock %}
+
+{% block list %}
+{% if item.hasChildren and options.depth is not sameas(0) and item.displayChildren %}
+    <ul{{ _self.attributes(listAttributes) }}>
+        {{ block('children') }}
+    </ul>
+{% endif %}
+{% endblock %}
+
+{% block children %}
+{# save current variables #}
+{% set currentOptions = options %}
+{% set currentItem = item %}
+{# update the depth for children #}
+{% if options.depth is not none %}
+{% set options = currentOptions|merge({'depth': currentOptions.depth - 1}) %}
+{% endif %}
+{% for item in currentItem.children %}
+    {% set active = false %}
+    {% if item.getExtra('admin') is not empty %}
+        {% if item.getExtra('admin').hasroute('list') and item.getExtra('admin').isGranted('LIST') and request.get('_sonata_admin') == item.getExtra('admin').code %}
+            {% set active = true %}
+        {% endif %}
+    {% else %}
+        {%  if item.route is defined and request.get('_route') == item.route %}
+            {% set active = true %}
+        {% endif %}
+    {% endif %}
+    {{ block('item', {'active': active}) }}
```

Comment:
This is currently unsupported statement, see https://github.com/twigphp/Twig/pull/1433